src/http2_stream.erl

```diff
@@ -574,10 +580,17 @@ check_content_length(Stream) ->
         undefined ->
             ok;
         _Other ->
-            case Stream#stream_state.request_body_size =:= binary_to_integer(ContentLength) of
-                true ->
-                    ok;
-                false ->
+            try binary_to_integer(ContentLength) of
+                Integer ->
+                    case Stream#stream_state.request_body_size =:= Integer of
+                        true ->
+                            ok;
+                        false ->
+                            rst_stream_(?PROTOCOL_ERROR, Stream),
+                            rst_stream
+                    end
+            catch
+                _:_ ->
```

Comment:
How is it dangerous? If the content length is bad, the result is the same no matter why it's bad.